projects/plugins/jetpack/modules/search/instant-search/store/reducer/query-string.js

```diff
@@ -79,3 +79,29 @@ export function filters( state = {}, action ) {
 
 	return state;
 }
+
+/**
+ * Reducer for keeping track of the user's selected static filter value
+ *
+ * @param {object} state - Current state.
+ * @param {object} action - Dispatched action.
+ *
+ * @returns {object} Updated state.
+ */
+export function staticFilters( state = {}, action ) {
```

Comment:
Could we possibly add tests for this in `instant-search/store/test/reducer.test.js`? I'd imagine it'd be pretty small and similar to [the sort reducer tests](https://github.com/Automattic/jetpack/blob/add/static-filters-support/projects/plugins/jetpack/modules/search/instant-search/store/test/reducer.test.js#L178).

Reply:
I've added a couple of them, but not sure how to mock the window allow list of static filter keys.

Reply:
@unDemian: Mocking the window object's get function ought to do the trick:

```js
test( 'mocking the window object for testing', () => {
  const exampleStaticFilters = [
    {
      filter_id: 'group_id',
      name: 'example',
      selected: null,
      type: 'group',
      values: [],
    },
  ];
  const windowSpy = jest.spyOn( window, 'window', 'get' );
  windowSpy.mockImplementation( () => ( {
    [ SERVER_OBJECT_NAME ]: {
      staticFilters: exampleStaticFilters,
    },
  } ) );
  expect( getAvailableStaticFilters() ).toEqual( exampleStaticFilters );
  windowSpy.mockRestore();
} );
```

Reply:
Awesome example, thank you! I've pushed some more tests.